packages/cubejs-server-core/src/core/OrchestratorApi.ts

```diff
@@ -129,7 +129,10 @@ export class OrchestratorApi {
         return res;
       }
 
-      data.dbType = await extractDbType(data);
+      if (!query.external) {
```

Comment:
Please pay attention on this line, `extractDbType` method use accessing the driver.
When we make request to external driver without `dataSource`, it will call regular driver that can lead to an exception because `dataSource` is empty.

Reply:
I'm not sure that is my specific case. 
If you will use the `executeQuery` method with `query.external = true` and not specify `dataSource` before then it will throw an Error inside the `extractDbType`. 
And as understand we do not need `dataSource` and dbType in the external case at all.

Reply:
As far as I remember, `query.external` flag could be set by the `QueryOrchestrator.getPreAggregationPreview` method only and only in a case when `PreAggregation` is configured as an `external` (it stores in an external data source not the Cubestore). The `OrchestrationApi.executeQuery` is called before this moment, so the `external` flag couldn't be true here.
On the other hand, I found what concerns me here: you call this method from the `ApiGateway.sqlRunner` method. However, we already have the `ApiGateway.getSqlResponseInternal` method designed for this. So, instead of producing a new pipeline for the query execution, I prefer to utilize the existing one. Please be able to change the `ApiGateway.sqlRunner` method in the following way: prepare the query object according to the `ApiGateway.getSqlResponseInternal` method, call it, and process the result according to your needs. This will keep the single query execution pipeline and disambiguate it.